### ISSUE TITLE
Orbital height as a game mechanic?

### ISSUE THREAD

**Issue by ajahraus** [TeamPorcupine/ProjectPorcupine#613]
I've been wondering, since it's a space themed game, should orbital height be a mechanic we should consider? 
For example, at the most basic level, I was thinking that a low orbit around a star may provide high amounts of solar energy, but very few other resources. Or maybe stars have distant asteroid fields, rich in minerals but low in sources of power. Maybe you have to be in orbit around a planet to trade at decent rates? Are there other star systems, if so, how is interstellar travel done? 
What are your thoughts?

**Comment by abackwood** [comment]
I'd love if you position in the world affected your station. Could be chosen at game start or maybe you can move your station with tons of fuel

**Comment by ajahraus** [comment]
@abackwood That's a good point, about the fuel. Should we use realistic orbital mechanics, or should be opt for a simpler version for the sake of game balance and pacing?

**Comment by mikejbrown** [comment]
Not sure if it fits the "base building game" atmos, but it does open some story possibilities. E.g. in addition to the stuff you mentioned already there could be a solar storm event with enough advanced warning that you can (if you choose) build engines to move out of the way... or build shields or do nothing and face the consequences depending on your play style.

**Comment by abackwood** [comment]
@mikejbrown yes exactly those kinds of things is what I'm hoping will be included. You make different choices based on where you are.

Other idea since you mentioned shields: one side of the world could be "towards the star". Building sheilds (or just walls i guess) can protect your station core from overheating. This isn't important in the habitable zone of the star, more so when close to the star. You end up with this assymetrical station design with sheilds on one side and infrastructure in the shade. Cool mental image :)

As for orbital mechanics: I think for this sort of game it's fine to fudge it a little. Amount of fuel needed could maybe be computed semi-realistically, but after that you either:
1) Consume a certain amount of fuel over time until the deltaV quota is reached (not quite realistic, but simpler)
2) Have designated points where you have to do burns. This is handled automatically for the player once they agree to the manouver (more realistic and provides more planning challenge, but complicated and has issues)

Questions I don't have an answer to:
- What happens if you run out of fuel before you're there? I don't think "you fly off into space and have to do corrections for more fuel later" is within the scope of this game. It's not KSP
- How intricate do the engines get? This is primarily a base-building game, not a ship simulator

**Comment by bjubes** [comment]
burns would be cool, but the option of messing up and crashing into your star and losing everything isn't exactly balanced.

**Comment by abackwood** [comment]
More likely youd end up in interplanetary space, far away from anything useful.

We're getting a bit away from the original topic: orbital parameters as a feature.
Changing orbits is also a kind of transit so the above still applies.

Reasons you might want to raise orbits
- Less time being obscured by the planet = more sunlight for solar panels
- Warping interstellar spacecraft face less disruption from the gravity well and have an easier time visiting your station = more and larger trade ships
- Less Kessler debris = less !!Fun!!

Reasons you might lower orbit:
- Easier for spacecraft to land = more small scale trade, easier planet mining, more visitors
- Within reach of planetbased weapons = more protection
- Prettier view for tourists/workers

**Comment by mikejbrown** [comment]
Instead of having orbital parameters that try to model real life more or less closely, what about a more abstracted region system roughly like this. I'm imagining the X and Y coordinates will just function a bit like the Rimworld world map: they are not really relevant to anything in game (and don't map to tiles in any direct way), but they may indirectly affect things like spawn rates, temperature etc. The simplest way to implement movement from a gameplay perspective might just be a sort of _jump gate_ style system in that you just click a button when you want to go somewhere and, if you have the resources in terms of fuel etc., you just pop up in the new region (effectively a save & reload operation with changing parameters).

```
<SolarSystem>
	<Region>
		<Name>Low Earth Orbit</Name>
		<Description>LEO: the hub of trade and commerce for the whole system</Description>
		<Location X="10" Y="0"/>
		<Params>
			<Param name="trader_chance" value="2.0"/> <!-- per hour or something...? -->
			<Param name="iron_asteroid_chance" value="0.1"/>
			<Param name="comet_chance" value="0"/>
			<Param name="pirate_chance" value="0.01"/>
        	<Param name="solar_power" value="1.0" />
        </Params>
	</Region>

	<Region>
		<Name>Main Belt</Name>
		<Description>Where all the belters are. You think you can smell them even through your spacesuit.</Description>
		<Location X="30" Y="0"/>
		<Params>
			<Param name="trader_chance" value="0.1"/> <!-- per hour or something...? -->
			<Param name="iron_asteroid_chance" value="5"/>
			<Param name="comet_chance" value="0"/>
			<Param name="pirate_chance" value="0.25"/>
        	<Param name="solar_power" value="0.75" />
        </Params>
	</Region>

	<Region>
		<Name>Kuiper Belt</Name>
		<Description>If you want to mine comets for water &amp; rare volatiles, you've come to the right place!</Description>
		<Location X="80" Y="0"/>
		<Params>
			<Param name="trader_chance" value="0"/> <!-- per hour or something...? -->
			<Param name="iron_asteroid_chance" value="0"/>
			<Param name="comet_chance" value="1"/>
			<Param name="pirate_chance" value="0.2"/>
        	<Param name="solar_power" value="0.2" />
        </Params>
	</Region>

	<Region>
		<Name>Tatooine</Name>
		<Description>Never a more wretched hive of scum and villainy</Description>
		<Location X="20" Y="0"/>
		<Params>
			<Param name="trader_chance" value="1.0"/> <!-- per hour or something...? -->
			<Param name="iron_asteroid_chance" value="0.5"/>
			<Param name="comet_chance" value="0"/>
			<Param name="pirate_chance" value="1.0"/>
        	<Param name="solar_power" value="1.5" />
        </Params>
	</Region>
</SolarSystem>
```

**Comment by mikejbrown** [comment]
Related issue #353

**Comment by GamerInspired** [comment]
I kinda like what @mikejbrown said  "what about a more abstracted region system", looking at the system he came up with in the example, I believe this would very moddable, and also allow different systems to have different things (if we have different systems) with different planet types and different chances for different parameters.

I think on discussion on fuel for moving around, it should maybe be something like if you don't have enough fuel you can't do the maneuver/jump. I also hope that if we are able to move around a system and to of stars then we have different engines/jump-drives that have different aspects; some maybe faster but require more fuel and other maybe be fuel efficient (require less fuel) but slower.

I would also think that having different maneuver/jump times (how long it takes to complete the action) would be a nice addiction; maybe having it so you people can move around but only inside a room (maybe moment/job speed is slower (can improve with tech? or maybe if you put your people in a cryopod the travel time will be instant; maybe machines that are automatic will be completed on arrival)). So the way I imagine this is that the game gabs everything that's connected to the craft with the engine/jump-drive and puts it on another "z-level" called "moving space" (or something) then it stays there until it arrives, which at that point it grabs it ad puts in on the new "level".

Of course with all ideas it will need more looking into and as a community we will need to decide.

Feedback is welcome.

**Comment by abackwood** [comment]
To be clear, I'm not talking about shifting the station up a few tile Z-levels. Shifting orbits would amount to an entirely new place for the station to be in. Station levels are in the order of meters, orbits in kilometers

**Comment by GamerInspired** [comment]
When I mentioned "z-level" I meant place. I wasn't quite sure what to say.
